angular/src/components/send/efflux-dates.component.ts

```diff
@@ -0,0 +1,325 @@
+import { DatePipe } from '@angular/common';
+import {
+    Directive,
+    EventEmitter,
+    Input,
+    OnInit,
+    Output
+} from '@angular/core';
+import { FormControl, FormGroup } from '@angular/forms';
+
+import { I18nService } from 'jslib-common/abstractions/i18n.service';
+import { PlatformUtilsService } from 'jslib-common/abstractions/platformUtils.service';
+
+// Different BrowserPath = different controls.
+enum BrowserPath {
+    // Native datetime-locale.
+    // We are happy.
+    Default = 'default',
+
+    // Native date and time inputs, but no datetime-locale.
+    // We use individual date and time inputs and create a datetime programatically on submit.
+    Firefox = 'firefox',
+
+    // No native date, time, or datetime-locale inputs.
+    // We use a polyfill for dates and a dropdown for times.
+    Safari = 'safari',
+}
+
+enum DateField {
+    DeletionDate = 'deletion',
+    ExpriationDate = 'expiration',
+}
+
+// Value = hours
+enum DatePreset {
+    OneHour = 1,
+    OneDay = 24,
+    TwoDays = 48,
+    ThreeDays = 72,
+    SevenDays = 168,
+    ThirtyDays = 720,
+    Custom = 0,
+    Never = null,
+}
+
+// TimeOption is used for the dropdown implementation of custom times
+// Standard = displayed time; Military = time used in logic
+interface TimeOption {
+    standard: string;
+    military: string;
```

Comment:
Small nit... Should we instead call this `twentyFourHour` or `24hour` if possible? Military time is a US construct in the auspice that only the US military uses it within the US, but the rest of the world commonly uses it and calls it 24-hour time or just "time".

Reply:
Changed to `twelveHour` and `twentyFourHour`, since `12hour`/`24hour` aren't an option.

I did learn that some folks call the 24 hour clock "railroad time" and in the U.K. it is called "continental time" sometimes, so there's some trivia for the future.